src/tools/generators.ts

```diff
@@ -354,6 +354,31 @@ export async function generateFromTemplate(
   return { written, exists, overwritten }
 }
 
+/**
+ * Checks a file for front matter.
+ */
+export function frontMatterDirectoryDir(generator: string): "string" | undefined {
+  if (!validateGenerator(generator)) {
+    return undefined
+  }
+
+  const { path } = filesystem
+
+  // where are we copying from?
+  const templateDir = path(templatesDir(), generator)
+  console.log(templateDir)
+
+  const fileContents = filesystem.read(`${templateDir}/NAME.tsx.ejs`)
+  const { data: frontMatterData } = frontMatter(fileContents)
+
+  if (frontMatterData?.destinationDir === undefined) {
+    warning(`⚠️  No front matter found in ${generator} template.`)
+    return undefined
+  }
```

Comment:
Do we always want to warn here? I maybe only want to see this if I haven't provided `--dir` because if that's my workflow and I never set it in the template, it's going to show each time.